src/skmatter/metrics/_prediction_rigidities.py

```diff
@@ -0,0 +1,199 @@
+import numpy as np
+from ..preprocessing import StandardFlexibleScaler
+
+def local_prediction_rigidity(
+    X_train,
+    X_test,
+    reg
+):
+    r"""Computes the local prediction rigidity (LPR) of a linear (kernel) model 
+    trained on a training dataset provided as input, on the local environments 
+    in the test set provided as a separate input. LPR is defined as follows:
+
+    .. math::
+
+        LPR_{i} = \frac{1}{X_i @ (XX + \lambda I)^{-1} @ X_i^{T}}
+
+    The function assumes that the model training is undertaken in a manner where the 
+    global prediction targets are averaged over the number of atoms appearing 
+    in each training structure, and the average feature vector of each structure
+    is hence used in the regression. This ensures that (1) regularization strength
+    across structures with different number of atoms is kept constant per structure
+    during model training, and (2) range of resulting LPR values are loosely kept 
+    between 0 and 1 for the ease of interpretation. This requires the user to provide 
+    the regularizer value that results from such training procedure. To guarantee valid
+    comparison in the LPR across different models, feature vectors are scaled
+    by a global factor based on standard deviation across atomic envs.
+
+    If the model is a kernel model, K_train and K_test can be provided in lieu of 
+    X_train and X_test, alnog with the appropriate regularizer for the trained model.
+    
+
+    Parameters
+    ----------
+    X_train : list of ndarray of shape (n_atoms, n_features)
+        Training dataset where each training set structure is stored as a separate 
+        ndarray.
+
+    X_test : list of ndarray of shape (n_atoms, n_features)
+        Test dataset where each training set structure is stored as a separate 
+        ndarray.        
+
+    reg : float value
+        Regularizer value that the linear/kernel model has been optimized to.
+
+    Returns
+    -------
+    LPR : list of arrays of shape (n_atoms)
+    rank_diff : integer value of the difference between cov matrix dimension and rank        
+
+    """
+
+    # initialize a StandardFlexibleScaler and fit to train set atom envs
+    scaler = StandardFlexibleScaler()
+    X_atom = np.vstack(X_train)
+    scaler.fit(X_atom)
+
+    # prep to build covariance matrix XX, take average feat vecs per struc
+    X_struc = []
+    for i, cur_X in enumerate(X_train):
+        X_struc.append(np.mean(scaler.transform(cur_X), axis=0))
+    X_struc = np.vstack(X_struc)
+
+    # build XX and obtain Xinv for LPR calculation
+    XX = X_struc.T @ X_struc
+    Xprime = XX + reg * np.eye(XX.shape[0])
+    rank_diff = X_struc.shape[1] - np.linalg.matrix_rank(Xprime)
+    Xinv = np.linalg.inv(Xprime)
```

Comment:
OK, will replace with pinv for now for improved numerical stability. This could be helpful especially in cases where the user, for example, wants a purely linear model w/o regularization. Need to experiment with this further.